### PR TITLE
Add podAntiAffinity rules

### PR DESCRIPTION
**Pull request by thepetk**
## What does this PR do?

The PR resolves the `kube-linter` error for `components/has/production` and `components/has/staging`:

```bash
<standard input>: (object: application-service/application-service-controller-manager apps/v1,
Kind=Deployment) pod's labels "control-plane=controller-manager" do not match with anti-affinity's
labels "app in (application-service-controller-manager)" (check: no-anti-affinity, remediation: Specify
anti-affinity in your pod specification to ensure that the orchestrator attempts to schedule replicas on
different nodes. Using podAntiAffinity, specify a labelSelector that matches pods for the deployment,
and set the topologyKey to kubernetes.io/hostname. Refer to https://kubernetes.io/docs/concepts/
scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity for details.)
```

In order to do so it adds a `preferredDuringSchedulingIgnoredDuringExecution` pod anti-affinity section which will try to match an `app` of value `application-service-controller-manager`. If it's not possible to match it will still schedule the pod in order to avoid any potential impact.

### Which issue(s) does this PR fix

fixes https://issues.redhat.com/browse/DEVHAS-543

### How to test changes / Special notes to the reviewer

By running locally the commands:
```
$ kustomize build components/has/staging/ | kube-linter lint -
$ kustomize build components/has/staging/ | kube-linter lint -
```

Both of them will have the response:
```
KubeLinter development

No lint errors found!
```

Other than that the changes introduced inside the `application-service-controller-manager` deployment will be this:
```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    control-plane: controller-manager
  name: application-service-controller-manager
  namespace: application-service
spec:
  replicas: 3
  selector:
    matchLabels:
      control-plane: controller-manager
  template:
    metadata:
      labels:
+      app: application-service-controller-manager
        control-plane: controller-manager
    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+            matchExpressions:
+             - key: app
+               operator: In
+               values:
+               - application-service-controller-manager
+           topologyKey: kubernetes.io/hostname
      containers:
...
```

### CODE REVIEW

File: components/has/k-components/replicas-patch/replicas_patch.yaml [hunk 1]
```diff
@@ -5,3 +5,20 @@ metadata:
...
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - application-service-controller-manager
+              topologyKey: kubernetes.io/hostname
```
Comment (hugares) [inline components/has/k-components/replicas-patch/replicas_patch.yaml:24]: nit: add a empty line at end of file

Comment (thepetk) [reply components/has/k-components/replicas-patch/replicas_patch.yaml]: Done :) Thx

**Review by hugares — APPROVED** [review]

**Review by johnmcollier — APPROVED** [review]
/lgtm
/approve

**Pull request merged**